src/main/scala/cognite/spark/v1/NumericDataPointsRelation.scala

```diff
@@ -36,11 +36,48 @@ case class InsertDataPointsItem(
     timestamp: Instant,
     value: Double)
 
+case class DataPointsDelete(
+    id: Option[Long],
+    externalId: Option[String],
+    inclusiveStart: Instant,
+    exclusiveEnd: Instant
+)
+
 class NumericDataPointsRelationV1(config: RelationConfig)(sqlContext: SQLContext)
-    extends DataPointsRelationV1[DataPointsItem](config)(sqlContext) {
+    extends DataPointsRelationV1[DataPointsItem](config)(sqlContext)
+    with WritableRelation {
 
   import PushdownUtilities.filtersToTimestampLimits
 
+  override def insert(rows: Seq[Row]): IO[Unit] = {
+    throw new RuntimeException("Insert not supported for datapoints. Use upsert instead.")
+  }
+
+  override def upsert(rows: Seq[Row]): IO[Unit] = insertSeqOfRows(rows)
+
+  override def update(rows: Seq[Row]): IO[Unit] = {
+    throw new RuntimeException("Update not supported for datapoints. Use upsert instead.")
+  }
+
+  override def delete(rows: Seq[Row]): IO[Unit] = {
+    val (dataPointsWithId, dataPointsWithExternalId) =
+      rows.map(r => fromRow[DataPointsDelete](r)).partition(p => p.id.isDefined)
```

Comment:
I'm a bit confused by this. What will this look like for the end user? How is this specified?

Reply:
I would have thought we'd need to get start and end from some filter.

Reply:
The user could do a query of the form "select id, externalId, inclusiveBegin, exclusiveEnd from table where id = timeSeriesId" and somehow specify a delete operation on the front-end

Reply:
The user can do a query specifying the time series id and / or external id, start and end of the time range to delete, and the table to delete from on some condition. When supported on the front end they could select delete as the operation and then just run the transform

Reply:
That's a strange and non-intuitive way of handling this. It's selecting column names that don't exist in the table, right? While technically possible and allowed by Spark (I guess? I find even that surprising, but I guess it doesn't check the schema at all) this is not how we should expose it, I think.

Something like "select idToDelete from datapoints where timestamp >= deleteFromHere and timestamp <= deleteUntilHere" would be more natural.

I don't know how easy that would be to implement, but I think we should perhaps leave deletes out of this for now.

Reply:
@Tapped any thoughts on what this would look like in Jetfire?

Reply:
removing delete for datapoints and stringdatapoints until we support deleting in jetfire. Need to come up with a good way to represent it.